cpp/src/arrow/compute/api_scalar.cc

```diff
@@ -344,6 +364,9 @@ static auto kRandomOptionsType = GetFunctionOptionsType<RandomOptions>(
     DataMember("length", &RandomOptions::length),
     DataMember("initializer", &RandomOptions::initializer),
     DataMember("seed", &RandomOptions::seed));
+static auto kMapArrayLookupOptionsType = GetFunctionOptionsType<MapArrayLookupOptions>(
```

Comment:
nit, but try to keep these organized alphabetically (though I see Random already broke that)